tests/test_client.py

```diff
@@ -167,6 +167,22 @@ def test_do_request_ok_yaml(self, fakesend, simple_config):
         # check we parsed the response
         assert ret == {"defaults": {"arm": {"machine": "ARM"}}}
 
+    @mock.patch("requests.sessions.Session.send", autospec=True)
+    def test_do_request_not_changed(self, fakesend, simple_config):
+        """Test do_request when receiving a 204 Not Changed reply"""
+        fakeresp = fakesend.return_value
+        fakeresp.status_code = 204
+        fakeresp.text = ""
+        client = oqc.OpenQA_Client()
+        request = requests.Request(
+            url=client.baseurl + "/api/v1/job_templates_scheduling/1", method="PUT"
+        )
+        ret = client.do_request(request)
+        # check we did not call .json() on the response (that would fail)
+        assert len(fakeresp.method_calls) == 0
```

Comment:
assert can take a second argument which I recommend to use, also see https://wiki.python.org/moin/UsingAssertionsEffectively , e.g.

```suggestion
        assert len(fakeresp.method_calls) == 0, 'invalid method call on response (or something)'
```